### DISCUSSION TITLE
How to set prefix key in Spacemacs?

### DISCUSSION THREAD

**Discussion by ghost** [syl20bnr/spacemacs#15735] [Q&A]
Hi,

I am using `rtags`, and want to customize the leading keys of `rtags` to 'M-RET t` by using

```
(rtags-enable-standard-keybindings c-mode-base-map "M-RET t")
```

How can I make Spacemacs display the key prefix `t` when typing `M-RET` as `rtags`? Currently, `t` is shown as `+prefix` when typing `M-RET`.

I tried the following: `(spacemacs/declare-prefix "M-RET t" "rtags")`, but it doesn't work.

Thank you!

**Comment by fiveNinePlusR** [comment]
maybe try `(spacemacs/declare-prefix (kbd "<M-RET> t") "rtags")`

**Comment by ghost** [comment]
Thanks for your suggestion. I just tried, but it didn't work either.

**Comment by ghost** [comment]
I ended up using this key sequence

`(which-key-add-key-based-replacements "<M-RET> t" "rtags-commands")`

If the prefix means only for some certain mode, then the following works:

`(which-key-add-major-mode-key-based-replacements 'c-mode "<M-RET> t" "rtags-commands")`

Maybe Spacemacs hasn't got any implementation of such key-based-replacements?

**Comment by mcp292** [comment]
If `(spacemacs/declare-prefix "M-RET t" "rtags")` doesn't work, `(spacemacs/declare-prefix "M-<return> t" "rtags")` should.

When passing in `KEY`, the name given by `describe-key`, before translation, should be used.

More binding functions can be found in [the spacemacs docs](https://www.spacemacs.org/doc/DOCUMENTATION.html#binding-keys) and in [this SO answer](https://stackoverflow.com/a/47094983/4682839).
